tests/function-controller/pkg/resource/resource.go

```diff
@@ -171,13 +171,10 @@ func (r *Resource) Update(res interface{}) (*unstructured.Unstructured, error) {
 	}
 
 	var result *unstructured.Unstructured
-	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		var errUpdate error
-		result, errUpdate = r.ResCli.Update(context.Background(), unstructuredObj, metav1.UpdateOptions{})
-		return errUpdate
-	}, r.log)
+
+	result, err = r.ResCli.Update(context.Background(), unstructuredObj, metav1.UpdateOptions{})
 	if err != nil {
-		return nil, errors.Wrapf(err, "while updating resource %s '%s'", r.kind, unstructuredObj.GetName())
+		return nil, err // upstream caller RetryOnConflict doesn't work with wrapped errors
```

Comment:
Can you give us the reason why?

Reply:
I didn't read it's actual code, but I think they don't bother unwrapping the error. The recommendation is documented there: https://github.com/kubernetes/client-go/blob/9927afa2880713c4332723b7f0865adee5e63a7b/util/retry/util.go#L89-L93

Reply:
Please add the link to those comments, it would make easier to understand why it behaves like that after some time.